backend/src/main/scala/cromwell/backend/BackendActor.scala

```diff
@@ -0,0 +1,85 @@
+package cromwell.backend
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import akka.event.LoggingReceive
+import cromwell.backend.BackendActor._
+import cromwell.backend.model.{JobDescriptor, WorkflowDescriptor}
+
+object BackendActor {
+
+  // Commands
+  sealed trait BackendActorMessage
+  case object BeforeAll extends BackendActorMessage
+  case class CreateJobExecutorActor(jobDescriptor: JobDescriptor) extends BackendActorMessage
+  case object AfterAll extends BackendActorMessage
+  case object Validate extends BackendActorMessage
+
+  // Events
+  sealed trait BackendActorEvent
+
+  sealed trait BeforeAllEvent extends BackendActorEvent
+  case object BeforeAllSucceeded extends BeforeAllEvent
+  case class BeforeAllFailed(throwable: Throwable) extends BeforeAllEvent
+
+  sealed trait JobExecutorCreationEvent extends BackendActorEvent
+  case class JobExecutorCreationSucceeded(backendJobExecutorActor: ActorRef) extends JobExecutorCreationEvent
+  case class JobExecutorCreationFailed(throwable: Throwable) extends JobExecutorCreationEvent
+
+  sealed trait AfterAllEvent extends BackendActorEvent
+  case object AfterAllSucceeded extends AfterAllEvent
+  case class AfterAllFailed(throwable: Throwable) extends AfterAllEvent
+
+  sealed trait ValidationEvent extends BackendActorEvent
+  case object ValidationSucceeded extends ValidationEvent
+  case class ValidationFailed(throwable: Throwable) extends ValidationEvent
+
+}
+
+/**
+  * Defines basic structure and functionality to initialize and make use of a backend through an Akka actor system.
+  * Backend functions should be implemented by each custom backend.
+  */
+trait BackendActor extends Actor with ActorLogging {
```

Comment:
@cjllanwarne ![Noooo](http://ct.fra.bz/ol/fz/sw/i57/5/5/24/frabz-One-does-not-simply-Like-his-own-comment-b046ac.jpg)